src/bokeh/document/callbacks.py

```diff
@@ -351,7 +351,8 @@ def trigger_event(self, event: Event) -> None:
                 if model:
                     model._trigger_event(event)
 
-        for cb in self._event_callbacks.get(event.event_name, []):
+        callbacks = self._event_callbacks.get(event.event_name, []).copy()
+        for cb in callbacks:
```

Comment:
What about 
```python
def callbacks_for_event(self, event_name: str) -> tuple[...]: 
    return tuple(self._event_callbacks.get(event.event_name, []))
```
Then we can have some basic unit tests for the immutable result and the "empty sequence" for missing events, and if we really wanted to pin this down, a mock/spy test to make sure that `trigger_event` uses the method instead of the private attr.